ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/BlurTransformation.kt

```diff
@@ -0,0 +1,338 @@
+ *                    Lower values produce a faster but lower quality blur.
+ */
+internal class BlurTransformation(
+    private val context: Context,
+    private val radius: Float = 25f,
+    private val scale: Float = 0.5f,
+) : Transformation {
+
+    override val cacheKey: String = "${javaClass.name}-$radius"
+
+    override suspend fun transform(
+        input: Bitmap,
+        size: Size,
+    ): Bitmap {
+        return input.blur(context, radius) ?: input
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        return other is BlurTransformation && radius == other.radius
+    }
+
+    override fun hashCode(): Int = radius.hashCode()
+}
+
+internal suspend fun Bitmap.blur(context: Context, radius: Float = 25f): Bitmap {
+    return if (Build.VERSION.SDK_INT >= 31) {
+        this.blur(context = context, radius = radius)
+    } else {
+        // useful for testing
+        val useRenderScript = true
+        if (useRenderScript) {
+            return blurUsingRenderScript(context, radius)
+        } else {
+            return blurByAveraging(scale = 1f, radius = radius.roundToInt())
+        }
+    }
+}
+
+internal fun Bitmap.blurUsingRenderScript(context: Context, radius: Float): Bitmap {
+
+    if (radius < 1f) {
+        return this@blurUsingRenderScript
+    }
+    // max radius supported by RenderScript is 25
+    val updatedRadius = min(radius.toDouble(), 25.toDouble())
+
+    val rs = RenderScript.create(context)
+    val input = Allocation.createFromBitmap(rs, this)
+    val output = Allocation.createTyped(rs, input.type)
+    val script = ScriptIntrinsicBlur.create(rs, Element.U8_4(rs))
+
+    script.setRadius(updatedRadius.toFloat())
+    script.setInput(input)
+    script.forEach(output)
+
+    val blurredBitmap = Bitmap.createBitmap(width, height, config)
+    output.copyTo(blurredBitmap)
+
+    input.destroy()
+    output.destroy()
+    script.destroy()
+    rs.destroy()
+
+    return blurredBitmap
+}
+
+/**
+ * Applies blur effect to a Bitmap.
+ *
+ * @param scale - The scale factor for resizing the image before blurring.
+ * @param radius - The radius of the square used for blurring.
+ * @return A new Bitmap with the blur effect applied.
+ *
+ * The algorithm is a blur transformation that applies a Gaussian blur effect to an input bitmap image.
+ * It works by computing the average color value of each pixel in a radius around the pixel.
+ * The radius of the blur effect is specified by the radius parameter.
+ *
+ * The algorithm uses a stack to keep track of the color values of the pixels in the blur radius.
+ * It iterates over each pixel in the image and computes the weighted sum of the red, green, and blue color values
+ * of the pixels in the blur radius.
+ *
+ * This sum is then divided by the sum of the divisor values to obtain the average color value for the pixel.
+ */
+internal suspend fun Bitmap.blurByAveraging(
```

Comment:
I tried to make this one more readable but it's still a mouthful. 
In a nutshell it goes pixel by pixel, counting the values of the pixels that are in a given radius, averages the number, then sets the average as the color